// broccoli-test-helper/dist/lib/output.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const path_1 = require("path");
const fixturify_1 = require("./fixturify");
const tree_diff_1 = require("./tree_diff");
class Output {
    constructor(builder) {
        this.builder = builder;
        this.treeDiff = new tree_diff_1.default(builder.outputPath);
    }
    read(from) {
        return fixturify_1.readSync(this.path(from));
    }
    path(subpath) {
        let outputPath = this.builder.outputPath;
        return subpath ? path_1.join(outputPath, subpath) : outputPath;
    }
    changes() {
        return this.treeDiff.changes;
    }
    build() {
        return this.builder.build().then(() => {
            this.treeDiff.recompute();
        });
    }
    /**
     * @deprecated
     */
    rebuild() {
        console.warn(`rebuilt() is deprecated, just call build() any.`);
        return this.build().then(() => this);
    }
    dispose() {
        return Promise.resolve(this.builder.cleanup());
    }
}
exports.default = Output;
